test/results/test_protein_folding_result.py

```diff
@@ -10,66 +10,130 @@
-        qubits compressed."""
-        lambda_back = 10
-        lambda_chiral = 10
-        lambda_1 = 10
-        penalty_terms = PenaltyParameters(lambda_chiral, lambda_back, lambda_1)
-
-        main_chain_residue_seq = "SAASSASAA"
-        side_chain_residue_sequences = ["", "", "A", "A", "A", "A", "A", "A", ""]
-
-        peptide = Peptide(main_chain_residue_seq, side_chain_residue_sequences)
-
-        mj_interaction = MiyazawaJerniganInteraction()
-
-        protein_folding_problem = ProteinFoldingProblem(peptide, mj_interaction, penalty_terms)
-        best_sequence = "101110010"
-
-        protein_folding_result = ProteinFoldingResult(protein_folding_problem, best_sequence)
-
-        result = protein_folding_result.get_result_binary_vector()
-
-        self.assertEqual(result, best_sequence)
 
-    def test_get_result_binary_vector_compressed(self):
-        """Tests if a protein folding result returns a correct expanded best sequence."""
-        lambda_back = 10
-        lambda_chiral = 10
-        lambda_1 = 10
-        penalty_terms = PenaltyParameters(lambda_chiral, lambda_back, lambda_1)
+def create_protein_folding_result(
+    main_chain: str, side_chains: List[str], best_sequence: str
+) -> ProteinFoldingResult:
+    """
+    Creates a protein_folding_problem, solves it and uses the result
+    to create a protein_folding_result instance.
+    Args:
+        main_chain: The desired main_chain for the molecules to be optimized
+        side_chains: The desired side_chains for the molecules to be optimized
+        best_sequence: The best sequence found by ProteinFoldingResult pre-computed
+    Returns:
+        Protein Folding Result
+    """
+    algorithm_globals.random_seed = 23
+    peptide = Peptide(main_chain, side_chains)
+    mj_interaction = MiyazawaJerniganInteraction()
 
-        main_chain_residue_seq = "SAASSASAA"
-        side_chain_residue_sequences = ["", "", "A", "A", "A", "A", "A", "A", ""]
+    penalty_back = 10
+    penalty_chiral = 10
+    penalty_1 = 10
 
-        peptide = Peptide(main_chain_residue_seq, side_chain_residue_sequences)
+    penalty_terms = PenaltyParameters(penalty_chiral, penalty_back, penalty_1)
 
-        mj_interaction = MiyazawaJerniganInteraction()
+    protein_folding_problem = ProteinFoldingProblem(peptide, mj_interaction, penalty_terms)
+    protein_folding_problem.qubit_op()
 
-        protein_folding_problem = ProteinFoldingProblem(peptide, mj_interaction, penalty_terms)
-        best_sequence = "101110010"
-        protein_folding_problem._unused_qubits = [0, 1, 2, 5, 7]
+    return ProteinFoldingResult(protein_folding_problem, best_sequence)
 
-        protein_folding_result = ProteinFoldingResult(protein_folding_problem, best_sequence)
 
-        result = protein_folding_result.get_result_binary_vector()
-        expected_sequence = "101110*0*10***"
+class TestProteinFoldingResult(QiskitNatureTestCase):
```

Comment:
I think testing every field in a separate method for a simple equality might be an overkill. We might test them in one test method.